### PR TITLE
Add lint

### PR DESCRIPTION
**Pull request by Shalinit3**
Added eslint(follows the rules outlined by 'airbnb') and husky.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      "react": {
+        "pragma": "React",
+        "version": "^16.4.2"
+      }
+    },
+    "parser": "babel-eslint",
+    "plugins": [
+      "react"
+    ],
+    "rules": {
```
Comment (shivangg) [inline .eslintrc:23]: Why not using the `airbnb` config?

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      "func-names": ["error", "never"],
+      "semi": ["error", "always"],
+      "quotes": ["error", "single"],
+      "camelcase": [ 1 ],
+      "no-undef": [ 1 ],
+      "react/jsx-curly-spacing": [ 2, "always" ],
+      "react/jsx-indent": [ 2, 4 ],
+      "react/prop-types": [ 1 ]
```
Comment (shivangg) [inline .eslintrc]: We should a popular style guide like `airbnb` and not write rules of our own. This will force us to kill our bad practices.

**Review by shivangg — CHANGES_REQUESTED** [review]
add airbnb config for eslint

File: .eslintrc (hunk 1 above)
Comment (Shalinit3) [reply .eslintrc:23]: We are using airbnb. Please check package.json

File: .eslintrc (hunk 2 above)
Comment (Shalinit3) [reply .eslintrc]: These are rules for react.

File: package.json [hunk 3]
```diff
@@ -4,6 +4,34 @@
...
+    "extends": "airbnb"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "**/*.{js,jsx,ts,tsx,json,css,scss,md}": [
+      "prettier --single-quote --write",
```
Comment (nimish-gupta) [inline package.json]: You can use prettier.rc for this

File: package.json [hunk 4]
```diff
@@ -4,6 +4,34 @@
   "private": true,
   "license": "MIT",
   "author": "Batch 5 Pesto",
+  "scripts": {
+    "lint": "eslint 'client/**/*.js' 'server/**/*.js' --ignore-pattern node_modules/",
```
Comment (nimish-gupta) [inline package.json]: add eslintrc file for this

File: .eslintrc (hunk 2 above)
Comment (nimish-gupta) [reply .eslintrc]: Why you added rules for `func-names`, `semi`, `quotes`, `camelcase`, `no-undef`?

**Review by nimish-gupta — CHANGES_REQUESTED** [review]

**Review by kaushikb11 — COMMENTED** [review]
Remove unnecessary lines from App.js.

File: .eslintrc (hunk 1 above)
Comment (Shalinit3) [reply .eslintrc:23]: Moved extension part to eslintrc

File: package.json (hunk 3 above)
Comment (Shalinit3) [reply package.json]: Moved this to prettierrc

File: package.json (hunk 4 above)
Comment (Shalinit3) [reply package.json]: Added this to eslintrc.

File: client/src/App.jsx [hunk 5]
```diff
@@ -9,7 +9,11 @@ class App extends Component {
         <header className="App-header">
           <img src={logo} className="App-logo" alt="logo" />
           <p>
-            Edit <code>src/App.js</code> and save to reload.
+            Edit
+            {' '}
```
Comment (nimish-gupta) [inline client/src/App.jsx:13]: Why are these present?

**Review by nimish-gupta — APPROVED** [review]

**Review by shivangg — APPROVED** [review]

**Pull request merged**